x-pack/plugins/cases/public/components/attachments/attachments_table.tsx

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useMemo, useState } from 'react';
+
+import type { EuiBasicTableColumn, Pagination, EuiBasicTableProps } from '@elastic/eui';
+import type { FileJSON } from '@kbn/shared-ux-file-types';
+
+import {
+  EuiModalBody,
+  EuiModalFooter,
+  EuiButtonIcon,
+  EuiLink,
+  EuiBasicTable,
+  EuiLoadingContent,
+  EuiSpacer,
+  EuiText,
+  EuiEmptyPrompt,
+  EuiButton,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+} from '@elastic/eui';
+import { useFilesContext } from '@kbn/shared-ux-file-context';
+import { FileImage } from '@kbn/shared-ux-file-image';
+
+import { APP_ID } from '../../../common';
+import { CASES_FILE_KINDS } from '../../files';
+
+interface AttachmentsTableProps {
+  isLoading: boolean;
+  items: FileJSON[];
+  onChange: EuiBasicTableProps<FileJSON>['onChange'];
+  pagination: Pagination;
+}
+
+export const AttachmentsTable = ({
+  items,
+  pagination,
+  onChange,
+  isLoading,
+}: AttachmentsTableProps) => {
+  const { client: filesClient } = useFilesContext();
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [selectedFile, setSelectedFile] = useState<FileJSON>();
+
+  const closeModal = () => setIsModalVisible(false);
+  const showModal = (file: FileJSON) => {
+    setSelectedFile(file);
+    setIsModalVisible(true);
+  };
+
+  const columns: Array<EuiBasicTableColumn<FileJSON>> = [
```

Comment:
Let's move the creation of the columns to a hook or a function outside of the component.

Reply:
> Let's move the creation of the columns to a hook

The `all cases list` does it in a hook but I never really understood why 😄

Reply:
For me, hooks are just functions that need to be aware of the lifecycle of the component. As we would create a function to improve readability we could do the same with a hook. If `columns` do not need to be aware of the lifecycle of the component (`isLoading`, etc) then it can be just a function. My suggestion is to avoid constructing the column inside the component to improve readability.